src/ng/q.js

```diff
@@ -354,8 +354,8 @@ function qFactory(nextTick, exceptionHandler) {
       var that = this;
       var done = false;
       try {
-        if ((isObject(val) || isFunction(val))) then = val && val.then;
-        if (isFunction(then)) {
+        if(isPromiseLike(val)) {
+          then = val && val.then
```

Comment:
Interestingly enough, I just re-read the spec notes and an accessor mutating the property is what they're trying to guard against, but that seems insane :speak_no_evil: 

In either case, changing this would result in the same number of `.then` property accesses but would be simpler.  The most correct thing would apparently to change `isPromiseLike` to return the `.then` property which you'd then `.call` with `val` here.  That'd be the most spec correct thing, but it just seems unnecessary.